pkg/devfile/adapters/kubernetes/adapter.go

```diff
@@ -0,0 +1,38 @@
+package kubernetes
+
+import (
+	"github.com/openshift/odo/pkg/devfile/adapters/common"
+	"github.com/openshift/odo/pkg/devfile/adapters/kubernetes/component"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/pkg/errors"
+)
+
+// Adapter maps Devfiles to Kubernetes resources and actions
+type Adapter struct {
+	componentAdapter common.ComponentAdapter
+	Client           kclient.Client
+	common.AdapterContext
+}
+
+// New instantiates a kubernetes adapter
+func New(adapterContext common.AdapterContext, client kclient.Client) Adapter {
+
+	compAdapter := component.New(adapterContext, client)
+
+	return Adapter{
+		componentAdapter: compAdapter,
+		AdapterContext:   adapterContext,
```

Comment:
Do we need `AdapterContext` here? As it is already a part of the `componentAdapter` as instantiated by `component.New()`

Reply:
I provided adapter context and client there in case they were needed at this higher level but you're right they're not currently needed so I'm fine removing them. Can add them if needed.